### ISSUE TITLE
conformance_install tests wedging

### ISSUE THREAD

**Issue by deads2k** [openshift/origin#18862]
Taking well over two hours

1. https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/8524/
2. https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/8537/
3. https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/8541/

At a glance

/assign @stevekuznetsov

**Comment by deads2k** [comment]
I've left https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/8537/ running

**Comment by stevekuznetsov** [comment]
/cc @derekwaynecarr @runcom @mrunalp 
Might be Docker related again? Similar symptoms to the `oci-umount` issues. Look at the artifacts to see installed packages. For instance, [this](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/8524/) job installed:

```
oci-umount.x86_64             2:2.3.3-3.gite3c9055.el7
```

**Comment by mrunalp** [comment]
I don't see oci-umount issues on @deads2k instance. Also, I was able to create a pod w/o any issues there.

**Comment by bparees** [comment]
since @derekwaynecarr asked me to look at this, here are my observations from the linked run:

(02:49:20 PM) bparees: decarr: i don't see any indication build_pruning.go failed?
(02:50:49 PM) bparees: decarr: I do see templateinstance readiness failed because it timed out waiting for a service acconut.
(02:52:18 PM) bparees: decarr: assuming that's the failure you want me to look at, see: Mar  6 16:35:29.822: INFO: Running 'oc get --config=/etc/origin/master/admin.kubeconfig --namespace=extended-test-templates-4cggx-j4c6g pods -o yaml'
(02:52:27 PM) bparees: decarr: we dump the build pod right there.
(02:52:53 PM) bparees: decarr: shows a container state of "running" for the main container.
(02:53:03 PM) bparees: decarr: and a few init containers that finished.
(02:53:42 PM) bparees: decarr: however the logs for the pod's main container show no output.
(02:54:03 PM) bparees: decarr: implying to me that it's not as running as it is claimed to be.
(02:54:14 PM) bparees: decarr: Mar  6 16:35:30.987: INFO: Log for pod "cakephp-mysql-example-1-build"/"sti-build"

tldr: seems like the build pod was created, the init containers ran, and when the main container was started, nothing happened (the pod reports the main container as running but there is no log output from that container and it never finishes)

**Comment by derekwaynecarr** [comment]
so looking at @deads2k job in question, we can see the following in build_pruning test.

1. at least one build runs to success myphp-build-1
2. the second build myphp-build-2 pod is created, each of the init containers runs to completion, and according to the kubelet, the main container is actually launched.  no log data is returned from that main container to let us know 'where' it is or is not stuck.

**Comment by bparees** [comment]
this PR (which we should not merge) will run the builds w/ very verbose logs.  Among other things, you'll see this as soon as the build container starts:
I0306 21:39:35.211810       1 builder.go:78] warning: OpenShift server version "v3.9.0-alpha.4+5ca3fd2-516" differs from this image "v3.9.0-alpha.4+9ab7a71"

or a similar message reporting the versions (if they match).

You'll see it multiple times, once from each init container and once from the main container.

https://github.com/openshift/origin/pull/18867

**Comment by derekwaynecarr** [comment]
similar timeout problem reported here for template readiness job:
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/8618/consoleFull

in this case, the main sti build container did start:

```
I0307 06:13:52.585545       1 docker.go:981] Creating container with options {Name:"s2i_centos_ruby_22_centos7_7d77a667" Config:{Hostname: Domainname: User: AttachStdin:false AttachStdout:true AttachStderr:false ExposedPorts:map[] Tty:false OpenStdin:true StdinOnce:true Env:[OPENSHIFT_BUILD_NAME=test-1 OPENSHIFT_BUILD_NAMESPACE=extended-test-build-timing-ndflt-5fs45 BUILD_LOGLEVEL=5] Cmd:[/bin/sh -c tar -C /tmp -xf - && /tmp/scripts/assemble] Healthcheck:<nil> ArgsEscaped:false Image:centos/ruby-22-centos7:latest Volumes:map[] WorkingDir: Entrypoint:[] NetworkDisabled:false MacAddress: OnBuild:[] Labels:map[] StopSignal: StopTimeout:<nil> Shell:[]} HostConfig:&{Binds:[] ContainerIDFile: LogConfig:{Type: Config:map[]} NetworkMode:container:048717a78375c8042be20477c0960936e45a433e7c52c75de34e0e17ca772764 PortBindings:map[] RestartPolicy:{Name: MaximumRetryCount:0} AutoRemove:false VolumeDriver: VolumesFrom:[] CapAdd:[] CapDrop:[KILL MKNOD SETGID SETUID] DNS:[] DNSOptions:[] DNSSearch:[] ExtraHosts:[] GroupAdd:[] IpcMode: Cgroup: Links:[] OomScoreAdj:0 PidMode: Privileged:false PublishAllPorts:false ReadonlyRootfs:false SecurityOpt:[] StorageOpt:map[] Tmpfs:map[] UTSMode: UsernsMode: ShmSize:67108864 Sysctls:map[] Runtime: ConsoleSize:[0 0] Isolation: Resources:{CPUShares:0 Memory:92233720368547 NanoCPUs:0 CgroupParent:kubepods-besteffort-podabe891c1_21ce_11e8_9f36_0eb54c40c92a.slice BlkioWeight:0 BlkioWeightDevice:[] BlkioDeviceReadBps:[] BlkioDeviceWriteBps:[] BlkioDeviceReadIOps:[] BlkioDeviceWriteIOps:[] CPUPeriod:0 CPUQuota:0 CPURealtimePeriod:0 CPURealtimeRuntime:0 CpusetCpus: CpusetMems: Devices:[] DeviceCgroupRules:[] DiskQuota:0 KernelMemory:0 MemoryReservation:0 MemorySwap:92233720368547 MemorySwappiness:<nil> OomKillDisable:<nil> PidsLimit:0 Ulimits:[] CPUCount:0 CPUPercent:0 IOMaximumIOps:0 IOMaximumBandwidth:0} Mounts:[] Init:<nil>}} ...
I0307 06:13:53.203050       1 docker.go:1013] Attaching to container "38e1bf102d36b144b1db41dbd7f98d2419f2d31dbae47628f82596a0f7ae741d" ...
I0307 06:13:53.217178       1 docker.go:1024] Starting container "38e1bf102d36b144b1db41dbd7f98d2419f2d31dbae47628f82596a0f7ae741d" ...
I0307 06:13:54.064686       1 tar.go:296] Adding to tar: /tmp/upload/scripts/assemble as scripts/assemble
I0307 06:13:54.064789       1 tar.go:296] Adding to tar: /tmp/upload/scripts/run as scripts/run
I0307 06:13:54.064856       1 tar.go:296] Adding to tar: /tmp/upload/src as src
I0307 06:13:54.064943       1 tar.go:296] Adding to tar: /tmp/upload/src/.s2i as src/.s2i
I0307 06:13:54.065003       1 tar.go:296] Adding to tar: /tmp/upload/src/.s2i/bin as src/.s2i/bin
I0307 06:13:54.065072       1 tar.go:296] Adding to tar: /tmp/upload/src/Gemfile as src/Gemfile
I0307 06:13:54.065143       1 tar.go:296] Adding to tar: /tmp/upload/src/config.ru as src/config.ru
<----end of log for "test-1-build"/"sti-build"
```

**Comment by bparees** [comment]
@derekwaynecarr yeah that looks like what gabe and I saw in our jenkins test job.  Our theory on that one is that the "38e1bf102d36b144b1db41dbd7f98d2419f2d31dbae47628f82596a0f7ae741d" container didn't really start, because we're streaming that tar input into it and something should be consuming that stream if it the container running.  (Also we'd expect to see the output from the entrypoint script that's being run in that container, if we got past the tar streaming): https://github.com/sclorg/s2i-ruby-container/blob/master/2.4/s2i/bin/assemble#L23)

**Comment by mrunalp** [comment]
Is this something that we can reproduce outside of the tests so we can look at the container that is getting stuck or not started?

**Comment by bparees** [comment]
@mrunalp based on what we're seeing in our jenkins job and this failure, just running an `s2i build  https://github.com/sclorg/ruby-ex centos/ruby-22-centos7` on one of our ec2 instances might be sufficient to produce the hang.  (you can get s2i from `github.com/openshift/source-to-image`)

**Comment by mrunalp** [comment]
@bparees okay, I will try that out.

**Comment by mrunalp** [comment]
@bparees I got 6 runs all successful in a row. I am keeping on trying them.

**Comment by mrunalp** [comment]
@bparees Maybe the layer that calls s2i build has an issue?

**Comment by bparees** [comment]
So here's what we've observed at this point:

1) with docker 1.12 and the latest s2i, "s2i build" does not appear to hang (@gabemontero please confirm)

2) with docker 1.13 and the 1.1.8 release of s2i, "s2i build" does not appear to hang.  - Note, origin-3.9 is using a version of s2i even older than this, thus I do not think origin-3.9 is susceptible to this issue...do we have evidence suggesting otherwise?

3) with docker 1.13 and the latest s2i code, "s2i build" hangs regularly.  The main change between 1.1.8 and 1.1.9 is we moved from the docker-engine api dependency to the docker client api dependency.  (see: https://github.com/openshift/source-to-image/pull/855 which appears to revert enough of the delta between 1.1.9 and 1.1.8 to make things work on docker 1.13 again).

So my current take is that (given/pending confirmation of(1)), something is still broken in docker 1.13 but somehow the change in client deps has exposed it.

**Comment by jim-minter** [comment]
https://github.com/moby/moby/issues/36516

**Comment by gabemontero** [comment]
On Wed, Mar 7, 2018 at 1:45 PM, Gabe Montero <gmontero@redhat.com> wrote:

>
>
> On Wed, Mar 7, 2018 at 12:54 PM, Ben Parees <notifications@github.com>
> wrote:
>
>> So here's what we've observed at this point:
>>
>>    1.
>>
>>    with docker 1.12 and the latest s2i, "s2i build" does not appear to
>>    hang (@gabemontero <https://github.com/gabemontero> please confirm)
>>
>> Yes I can confirm.  Many iterations of running s2i build from the jenkins
> image test with an `s2i` built that uses docker client vs. docker-engine
> using docker 1.12.6
> on an ami instance I provisioned before the change in our openshift ami's.
>
> FWIIW, I'm running 1.13.1 on my laptop and have not see the problem after
> many attempts with the "docker client" s2i either.  I've rebuilt s2i
> several times in the
> last week, most recently March 6th after rebasing.
>

On the jenkins test/merge CI jobs, there have been streaks where we hit
this, then streaks where we do not, since the AMI docker bump to 1.13

**Comment by gabemontero** [comment]
On Wed, Mar 7, 2018 at 12:54 PM, Ben Parees <notifications@github.com>
wrote:

> So here's what we've observed at this point:
>
>    1.
>
>    with docker 1.12 and the latest s2i, "s2i build" does not appear to
>    hang (@gabemontero <https://github.com/gabemontero> please confirm)
>
> Yes I can confirm.  Many iterations of running s2i build from the jenkins
image test with an `s2i` built that uses docker client vs. docker-engine
using docker 1.12.6
on an ami instance I provisioned before the change in our openshift ami's.

FWIIW, I'm running 1.13.1 on my laptop and have not see the problem after
many attempts with the "docker client" s2i either.  I've rebuilt s2i
several times in the
last week, most recently March 6th after rebasing.

**Comment by bparees** [comment]
for master this should be fixed by https://github.com/openshift/origin/pull/18882
